src/main/java/com/google/cloud/pubsublite/spark/PslMicroBatchInputPartitionReader.java

```diff
@@ -64,7 +64,13 @@ public boolean next() {
         msg = subscriber.messageIfAvailable();
         break;
       } catch (TimeoutException e) {
-        log.atWarning().log("Unable to get any messages in last " + SUBSCRIBER_PULL_TIMEOUT);
+        log.atWarning().log(
+            String.format(
+                "Unable to get any messages in last %s. Partition: %d; Current message offset: %d; End message offset: %d.",
+                SUBSCRIBER_PULL_TIMEOUT.toString(),
+                endOffset.partition().value(),
+                currentMsg != null ? currentMsg.offset().value() : null,
```

Comment:
nit. if using both branches, test the positive first. I.e.

currentMsg == null ? null : ...